src/CalcViewModel/UnitConverterViewModel.cpp

```diff
@@ -293,7 +319,8 @@ String ^ UnitConverterViewModel::ConvertToLocalizedString(const std::wstring& st
         std::wstring significandStr(stringToLocalize.substr(0, posOfE));
         std::wstring exponentStr(stringToLocalize.substr(posOfSign + 1, stringToLocalize.length() - posOfSign));
 
-        result += ConvertToLocalizedString(significandStr, allowPartialStrings) + "e" + signOfE + ConvertToLocalizedString(exponentStr, allowPartialStrings);
+        result += ConvertToLocalizedString(significandStr, allowPartialStrings, cfp) + "e" + signOfE
```

Comment:
There is a bug when testing with a currency without fractional digits, I get two different answers in this circumstance:

1. Switch first currency to "Test Fractional Digits"
2. Switch second currency to "Test Fractional Digits"
3. In the first currency input, type 1.999
4. Switch the first currency to "Test No Fractional Digits" 
5. Notice the conversion says 2 -> 399.800, even though the conversion rate is 1 = 200.00
6. Hit the CE buton, type 2 in manually, notice the answer is now 2 -> 400.

I think the issue is that we display how the currency is formatted and displayed but we don't use that value in the actual calculation.

Reply:
Wow, thanks for your finding! 
Seems that I have to truncate the digits after switching the currency, which means the digits will be gone forever even if I switch back.

Reply:
Fixed and added a UI test case.